accepted/00000-ptf-repos-in-uyuni.md

```diff
@@ -0,0 +1,162 @@
+To identify them as PTF repositories we can check if the path start with `PTF/Release`.
+
+If they are PTF repositories, we need to generate the missing values for the Channel mapping like this:
+
+- scc repository id: take it from the repository definition
+- scc product id: search for the product using `Product Identifier`, `Product Version`, and `Product Architecture` like specified in the URL
+- scc root product id: find all existing root products for the product id. We need to generate an entry for all of them
+
+The values for update tag and mandatory can be hard coded as they are the same 
+for all PTF repositories.
+
+- update tag: empty [1]
+- mandatory: false
+
+As PTF repositories do not get patches, an update tag is not needed.
+The mandatory flag should be `false` as PTF repositories should only be used when needed.
+
+As we need to generate entries for all existing root products we can iterate over all we found.
+For every root product we take the first channel we can find which does have a `parent channel label` set.
+We can take this as parent channel label for our new entry.
+
+As the channel label and name must be unique.
+We can generate the label and name from the URL components:
+
+    <Customer Account ID> <Product Identifier> <Product Version> "PTFs" ["Debuginfo"] <Product Architecture>
+
+The label is computed using the lowercase components joined with a dash ("-") and the name just joined with a space (" ").
+
+This can either be done by Uyuni Server or SCC. In case SCC use this algorithm it could set it as `repository name`
+joined with a dash ("-") and in `repository description` using spaces.
+
+The product tree uses additional suffixes for products which belongs to multiple root products to make label and name unique.
+The suffix is appended always after the architecture of the channel label.
+We can parse it from the entry we used to find the `parent channel label` and use it to enhance the component list.
+
+
+## Test Package Channels
+
+Before generating PTFs, SUSE will first create so called TEST packages.
+The customer need to test them and report positive result before SUSE make an official supported PTF out of it.
+
+TEST packages should only be installed on dedicated test systems.
+Therefor it is important that they get not installed on any system where just the version of the TEST package is higher than the installed one.
+
+Current idea is, that TEST packages will go into a different channel. Probably the last directory will be names "test" and "test_debug".
+We would use the same rules as for PTF Channels and adjust the name and label.
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- we are guessing the values. Some, like the channel label, are sensitive and must not be changed. We have to be sure that the rules and the structure never changes.
+- ptf handling for non SUSE OSes will not be as nice as for SUSE OSes.
```

Comment:
Can you elaborate on what will not be nice?

Reply:
done